### ISSUE TITLE
[Bug] aptos-node gives misleading error message when config points to non-existing genesis blob/waypoint or data dir.

### ISSUE THREAD

**Issue by geekflyer** [aptos-labs/aptos-core#2399]
When trying to start an aptos-node with a fullnode config file, which points in turn to a genesis blob that does not exist, the error message makes it sound as-if the fullnode config file cannot be found, which is pretty misleading.

Repro:

```
cargo run --bin aptos-node -- -f docker/compose/public_full_node/public_full_node.yaml
```

fails with:
```
thread 'main' panicked at 'Failed to load node config file! Given file path: "docker/compose/public_full_node/public_full_node.yaml". Error: IO("genesis", Os { code: 2, kind: NotFound, message: "No such file or directory" })', aptos-node/src/lib.rs:136:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

My _guess_ is this is caused by the error wrapping here (wraps things simply with `genesis`) which masks the original error:

https://github.com/aptos-labs/aptos-core/blob/072e7e0f3917335ae97ca202700b7b99418e75c5/config/src/config/execution_config.rs#L56-L84

We might also want to change the `Failed to load node config file` to distinguish between a file not having been found and a file containing invalid configuration values (such as pointing to a genesis blob that doesn't exist).

**Comment by geekflyer** [comment]
Btw, getting the _exact_ same error message when the waypoint or data dir doesn't exist as well. So it's really hard to tell from that error message alone what's wrong with the config.

**Comment by davidiw** [comment]
Ahh... good times. thanks for looking into this.

**Comment by wrwg** [comment]
This is not a wrapping problem. You actually see the underlying error: `IO("genesis", Os { code: 2, kind: NotFound, message: "No such file or directory" }(`. It's just that by default Rust does not print the file name on this IO error. One need to go to [here](https://github.com/aptos-labs/aptos-core/blob/175aeb3935d7877e2b06c57eda284afcb4388977/aptos-move/framework/src/release_bundle.rs#L53) and add an anyhow `with_context` or such.

**Comment by wrwg** [comment]
I think the above PR closes this, if not reopen.

**Comment by geekflyer** [comment]
@wrwg I know it shows the underlying error but due to the fact that simply adds a `genesis` (without any further explanation) combined with a top level error message that points to a concrete, but valid file path, makes it very confusing.

I think your PR fixes a similar issue when loading move release bundles, but my issue was mainly about this occuring when loading the node config file (in particular the genesis and waypoint file paths in it) which I don't think hasn't been touch/fixed with #3350. 

Therefore reopening :)

**Comment by gregnazario** [comment]
Looks like this was cleaned up
